nav2_smoother/src/nav2_smoother.cpp

```diff
@@ -0,0 +1,298 @@
+  auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+      get_node_base_interface(), get_node_timers_interface());
+  tf_->setCreateTimerInterface(timer_interface);
+  transform_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_);
+
+  std::string costmap_topic, footprint_topic;
+  double transform_tolerance;
+  this->get_parameter("costmap_topic", costmap_topic);
+  this->get_parameter("footprint_topic", footprint_topic);
+  this->get_parameter("transform_tolerance", transform_tolerance);
+  costmap_sub_ = std::make_shared<nav2_costmap_2d::CostmapSubscriber>(
+      shared_from_this(), costmap_topic);
+  footprint_sub_ = std::make_shared<nav2_costmap_2d::FootprintSubscriber>(
+      shared_from_this(), footprint_topic, 1.0);
+
+  std::string global_frame, robot_base_frame;
+  get_parameter("global_frame", global_frame);
+  get_parameter("robot_base_frame", robot_base_frame);
+  collision_checker_ =
+      std::make_shared<nav2_costmap_2d::CostmapTopicCollisionChecker>(
+          *costmap_sub_, *footprint_sub_, *tf_, this->get_name(), global_frame,
+          robot_base_frame, transform_tolerance);
+
+  if (!loadSmootherPlugins()) {
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  // Initialize pubs & subs
+  plan_publisher_ = create_publisher<nav_msgs::msg::Path>("plan_smoothed", 1);
+
+  // Create the action server that we implement with our smoothPath method
+  action_server_ = std::make_unique<ActionServer>(
+      rclcpp_node_, "smooth_path",
+      std::bind(&SmootherServer::smoothPlan, this));
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+bool SmootherServer::loadSmootherPlugins() {
+  auto node = shared_from_this();
+
+  smoother_types_.resize(smoother_ids_.size());
+
+  for (size_t i = 0; i != smoother_ids_.size(); i++) {
+    try {
+      smoother_types_[i] =
+          nav2_util::get_plugin_type_param(node, smoother_ids_[i]);
+      nav2_core::Smoother::Ptr smoother =
+          lp_loader_.createUniqueInstance(smoother_types_[i]);
+      RCLCPP_INFO(get_logger(), "Created smoother : %s of type %s",
+                  smoother_ids_[i].c_str(), smoother_types_[i].c_str());
+      smoother->configure(node, smoother_ids_[i], tf_, costmap_sub_,
+                          footprint_sub_);
+      smoothers_.insert({smoother_ids_[i], smoother});
+    } catch (const pluginlib::PluginlibException &ex) {
+      RCLCPP_FATAL(get_logger(), "Failed to create smoother. Exception: %s",
+                   ex.what());
+      return false;
+    }
+  }
+
+  for (size_t i = 0; i != smoother_ids_.size(); i++) {
+    smoother_ids_concat_ += smoother_ids_[i] + std::string(" ");
+  }
+
+  RCLCPP_INFO(get_logger(), "Smoother Server has %s smoothers available.",
+              smoother_ids_concat_.c_str());
+
+  return true;
+}
+
+nav2_util::CallbackReturn
+SmootherServer::on_activate(const rclcpp_lifecycle::State &) {
+  RCLCPP_INFO(get_logger(), "Activating");
+
+  RCLCPP_INFO(get_logger(), "Checking transform");
```

Comment:
I think this log and the `Rate` object can be removed? Looks like an old debug artifact